crates/dojo-lang/src/compiler.rs

```diff
@@ -131,8 +143,37 @@ impl Compiler for DojoCompiler {
         };
 
         let mut compiled_classes: HashMap<String, (Felt, ContractClass)> = HashMap::new();
+        let list_selector =  get_allowed_libfuncs_list(&props, &unit);
 
         for (decl, class) in zip(contracts, classes) {
```

Comment:
```suggestion
        for (decl, class) in zip(contracts, classes) {
            let contract_name = decl.submodule_id.name(db.upcast_mut());

            // note that the qualified path is in snake case while
            // the `full_path()` method of StructId uses the original struct name case.
            // (see in `get_dojo_model_artifacts`)
            let qualified_path = decl.module_id().full_path(db.upcast_mut());
```
Let's move the `contract_name` and `qualified_path` here to be re-used in error messages.